composer/utils/inference.py

```diff
@@ -180,11 +180,17 @@ def export_for_inference(
                     raise ValueError(f'sample_input argument is required for onnx export')
                 sample_input = ensure_tuple(sample_input)
 
+                input_names = ['input']
+
+                # Extract input names from sample_input if it is a dict
+                if isinstance(sample_input[0], dict):
```

Comment:
Not sure if HF (or others) Models have a combination of keyword vs non-keyword arguments. For just keyword or just non-keyword, this would work.

Reply:
Is there any way to make this robust?

Reply:
Iterating over `sample_input` and extracting the first dict arg except when it's last?

Reply:
Done. I extract all dictionary arguments for named inputs.